lib/ansible/modules/cloud/misc/terraform.py

```diff
@@ -177,7 +177,8 @@ def preflight_validation(bin_path, project_path, variables_args=None, plan_file=
     if not os.path.isdir(project_path):
         module.fail_json(msg="Path for Terraform project '{0}' doesn't exist on this host - check the path and try again please.".format(project_path))
 
-    rc, out, err = module.run_command([bin_path, 'validate'] + variables_args, cwd=project_path)
+    cmd_str = " ".join([bin_path, 'validate'] + variables_args)
```

Comment:
you should not do it this way, as this will break if arguments need quoting, also know that passing a single string as a command can be problematic as the whole thing is interpreted as the command.

If you really need this, set `use_unsafe_shell=true` on run_command and it will properly stringify everything and execute it via `/bin/sh`.

Reply:
The other option is to strip the arguments and keep it as a list